### PR TITLE
PeerTest fix

### PR DESCRIPTION
**Pull request by arndey**
Fixes for tests about topology changes

### CODE REVIEW

**Comment by mversic** [comment]
can you clear up commit history?

**Review by mversic — CHANGES_REQUESTED** [review]

File: modules/test-tools/src/main/kotlin/jp/co/soramitsu/iroha2/testengine/IrohaContainer.kt [hunk 1]
```diff
@@ -32,11 +31,14 @@ open class IrohaContainer : GenericContainer<IrohaContainer> {
     constructor(config: IrohaConfig.() -> Unit = {}) : this(IrohaConfig().apply(config))
 
     constructor(config: IrohaConfig) : super(
-        DockerImageName.parse("${config.imageName}:${config.imageTag}")
+        DockerImageName.parse("${config.imageName}:${config.imageTag}"),
     ) {
         val publicKey = config.keyPair.public.bytes().toHex()
         val privateKey = config.keyPair.private.bytes().toHex()
```
[the remark this answers is not in the document]
Comment (arndey) [reply modules/test-tools/src/main/kotlin/jp/co/soramitsu/iroha2/testengine/IrohaContainer.kt:37]: no

File: modules/client/src/test/kotlin/jp/co/soramitsu/iroha2/testengine/Genesis.kt [hunk 2]
```diff
@@ -29,19 +29,29 @@ import org.testcontainers.shaded.org.apache.commons.lang3.RandomStringUtils
  */
 open class DefaultGenesis : Genesis(rawGenesisBlock())
 
+open class AliceCanUnregisterAnyPeer : Genesis(
```
[the remark this answers is not in the document]
Comment (arndey) [reply modules/client/src/test/kotlin/jp/co/soramitsu/iroha2/testengine/Genesis.kt:32]: to give an option to inherit from this class
sometimes Genesis classes can be inherited not from DefaultGenesis

File: modules/test-tools/src/main/kotlin/jp/co/soramitsu/iroha2/testengine/IrohaContainer.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (arndey) [reply modules/test-tools/src/main/kotlin/jp/co/soramitsu/iroha2/testengine/IrohaContainer.kt:37]: generated for each container

File: modules/test-tools/src/main/kotlin/jp/co/soramitsu/iroha2/testengine/IrohaRunnerExtension.kt [hunk 3]
```diff
@@ -175,6 +175,7 @@ class IrohaRunnerExtension : InvocationInterceptor, BeforeEachCallback {
             keyPairs.add(generateKeyPair())
             portsList.add(findFreePorts(3)) // P2P + API + TELEMETRY
         }
+        val genesisKeyPair = keyPairs.first()
```
Comment (mversic) [inline modules/test-tools/src/main/kotlin/jp/co/soramitsu/iroha2/testengine/IrohaRunnerExtension.kt:178]:
```suggestion
        val genesisKeyPair = generateKeyPair()
```

how about this? we shouldn't be using the same keys for nodes and genesis

File: modules/test-tools/src/main/kotlin/jp/co/soramitsu/iroha2/testengine/IrohaRunnerExtension.kt [hunk 4]
```diff
@@ -192,6 +193,7 @@ class IrohaRunnerExtension : InvocationInterceptor, BeforeEachCallback {
                     }
                     alias = IrohaContainer.NETWORK_ALIAS + p2pPort
                     keyPair = keyPairs[n]
+                    this.genesisKeyPair = genesisKeyPair
```
Comment (mversic) [inline modules/test-tools/src/main/kotlin/jp/co/soramitsu/iroha2/testengine/IrohaRunnerExtension.kt:196]: why do you have to use `this` here?

File: modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/client/Iroha2Client.kt [hunk 5]
```diff
@@ -85,7 +85,7 @@ open class Iroha2Client(
     open val credentials: String? = null,
     open val eventReadTimeoutInMills: Long = 250,
     open val eventReadMaxAttempts: Int = 10,
-    override val coroutineContext: CoroutineContext = Dispatchers.IO + SupervisorJob()
+    override val coroutineContext: CoroutineContext = Dispatchers.IO + SupervisorJob(),
```
Comment (Mingela) [inline modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/client/Iroha2Client.kt:88]: haven't we agreed to not use this code style?

**Review by Mingela — APPROVED** [review]

**Comment by arndey** [comment]
> can you clear up commit history?

For what? With use squash

**Review by Mingela — APPROVED** [review]

File: modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/client/Iroha2Client.kt (hunk 5 above)
Comment (arndey) [reply modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/client/Iroha2Client.kt:88]: In my opinion it make no sense to resist it (this code style option)

Comment (arndey) [reply modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/client/Iroha2Client.kt:88]: as I understand it, it is accepted by most of the community

**Review by gv-timur — APPROVED** [review]

**Review by mversic — APPROVED** [review]

**Review by Mingela — APPROVED** [review]

**Pull request merged**